tests/lib/c_lib/src/main.c

```diff
@@ -514,38 +502,96 @@ void test_checktype(void)
 }
 
 /**
+ * @brief Test memchr function
  *
- * @brief Test character operation
- *
- * @see memchr(), memcpy(), memmove().
- *
+ * @see memchr().
  */
-void test_memstr(void)
+void test_memchr(void)
 {
-	int i, j, ret;
 	static const char str[] = "testfunction";
-	char   arr[100], move_arr[] = "12123";
-	static const char num[] = "1234567891234567891234";
-
-	zassert_is_null(memchr(str, 'a', 0), "memchr 0 error");
-	zassert_not_null(memchr(str, 'e', 10), "memchr serach e");
-	zassert_is_null(memchr(str, 'e', 1), "memchr e error");
-
-	for (i = 0; i < 20; i++) {
-		for (j = 0; j < 20; j++) {
-			memcpy(&arr[i], num, 0);
-			ret = memcmp(&num[j], &arr[i], 0);
-			zassert_true((ret == 0), "memcpy failed");
-			memcpy(&arr[i], &num[j], 1);
-			ret = memcmp(&num[j], &arr[i], 1);
-			zassert_true((ret == 0), "memcpy failed");
```

Comment:
Hi, I remember the previous code include this before @xiaofst modified it. Could you please take a look the original code before his PR? I'm afraid of something missing, thanks!

Reply:
Hi, enjia. This code you mentioned was writen by Jian last year and xiaofs improve the coverage on the basis of Jian's code. I redesigned this testcase, now the coverage of memchr(), memcpy(), memmove() is also 100%.